src/extension/iSamples_resultList.js

```diff
@@ -36,8 +37,32 @@ class ResultList extends React.Component {
     this.props.setView({ ...store.getState()['query']['view'], facet: format });
   }
 
+  spinner = () => {
+    let color = "slategray";
+    const override = {
+      display: "flex",
+      height: "100%",
+    };
+    return (
+      <div style={{
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        margin:"2em",
+    }}>
+        <FadeLoader
+        color={color}
+        loading={true}
+        cssOverride={override}
+        aria-label="Loading Spinner"
+        data-testid="loader"
+        />
+      </div>
+    );   
+  }
+  
   render() {
-    const { bootstrapCss, onChange, setView } = this.props;
+    const { bootstrapCss, onChange, setView, pending} = this.props;
```

Comment:
pending variable indicates whether we are in progress of fetching the results. We get this state from the redux store.